sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_client_base.py

```diff
@@ -0,0 +1,81 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import TYPE_CHECKING
+
+from azure.core.pipeline import AsyncPipeline
+
+from . import AsyncChallengeAuthPolicy
+from .client_base import _get_policies
+from .multi_api import load_generated_api
+
+if TYPE_CHECKING:
+    try:
+        # pylint:disable=unused-import
+        from typing import Any
+        from azure.core.configuration import Configuration
+        from azure.core.pipeline.transport import AsyncHttpTransport
+        from azure.core.credentials_async import AsyncTokenCredential
+    except ImportError:
+        # AsyncTokenCredential is a typing_extensions.Protocol; we don't depend on that package
+        pass
+
+
+def _build_pipeline(config: "Configuration", transport: "AsyncHttpTransport" = None, **kwargs: "Any") -> AsyncPipeline:
+    policies = _get_policies(config, **kwargs)
+    if transport is None:
+        from azure.core.pipeline.transport import AioHttpTransport
+
+        transport = AioHttpTransport(**kwargs)
+
+    return AsyncPipeline(transport, policies=policies)
+
+
+class AsyncKeyVaultClientBase(object):
+    def __init__(self, vault_url: str, credential: "AsyncTokenCredential", **kwargs: "Any") -> None:
+        if not credential:
+            raise ValueError(
+                "credential should be an object supporting the AsyncTokenCredential protocol, "
+                "such as a credential from azure-identity"
+            )
+        if not vault_url:
+            raise ValueError("vault_url must be the URL of an Azure Key Vault")
+
+        self._vault_url = vault_url.strip(" /")
+        client = kwargs.get("generated_client")
+        if client:
+            # caller provided a configured client -> nothing left to initialize
+            self._client = client
+            return
+
+        api_version = kwargs.pop("api_version", None)
```

Comment:
Since this package is currently just 7.2-preview, should we not include checking api_version as a kwarg and calling `load_generated_api` from the multi_api file?

My thinking is that for now we just only load the 7.2-preview code, and by the time there is a new service api version for administration, we can generate the code with the flag `--multiapi` and autorest will generate a multiapi client for us, and then we can pass in api_version directly to the generated multiapi client